---
title: "class 15"
author: "Ramola Baviskar (PID A12228297)"
date: "3/8/2022"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Install datapasta

#Investigate Pertussis case numbers over time in the US

The CDC has tracked case numbers since the early 1920s. 
https://www.cdc.gov/pertussis/surv-reporting/cases-by-year.html

```{r}
cdc <- data.frame(
                                 Year = c(1922L,1923L,1924L,1925L,
                                          1926L,1927L,1928L,1929L,1930L,1931L,
                                          1932L,1933L,1934L,1935L,1936L,
                                          1937L,1938L,1939L,1940L,1941L,1942L,
                                          1943L,1944L,1945L,1946L,1947L,
                                          1948L,1949L,1950L,1951L,1952L,
                                          1953L,1954L,1955L,1956L,1957L,1958L,
                                          1959L,1960L,1961L,1962L,1963L,
                                          1964L,1965L,1966L,1967L,1968L,1969L,
                                          1970L,1971L,1972L,1973L,1974L,
                                          1975L,1976L,1977L,1978L,1979L,1980L,
                                          1981L,1982L,1983L,1984L,1985L,
                                          1986L,1987L,1988L,1989L,1990L,
                                          1991L,1992L,1993L,1994L,1995L,1996L,
                                          1997L,1998L,1999L,2000L,2001L,
                                          2002L,2003L,2004L,2005L,2006L,2007L,
                                          2008L,2009L,2010L,2011L,2012L,
                                          2013L,2014L,2015L,2016L,2017L,2018L,
                                          2019L),
         No..Reported.Pertussis.Cases = c(107473,164191,165418,152003,
                                          202210,181411,161799,197371,
                                          166914,172559,215343,179135,265269,
                                          180518,147237,214652,227319,103188,
                                          183866,222202,191383,191890,109873,
                                          133792,109860,156517,74715,69479,
                                          120718,68687,45030,37129,60886,
                                          62786,31732,28295,32148,40005,
                                          14809,11468,17749,17135,13005,6799,
                                          7717,9718,4810,3285,4249,3036,
                                          3287,1759,2402,1738,1010,2177,2063,
                                          1623,1730,1248,1895,2463,2276,
                                          3589,4195,2823,3450,4157,4570,
                                          2719,4083,6586,4617,5137,7796,6564,
                                          7405,7298,7867,7580,9771,11647,
                                          25827,25616,15632,10454,13278,
                                          16858,27550,18719,48277,28639,32971,
                                          20762,17972,18975,15609,18617)
       )
```

#Now use ggplot.
```{r}
library(ggplot2)
library(tidyverse)
```

>Q1. 
>Q2.

```{r}
Pertussis_linegraph <- ggplot(cdc) + 
  aes(Year, No..Reported.Pertussis.Cases) +
  geom_point() +
  geom_line() 
```

```{r}
Pertussis_linegraph +
  geom_vline(xintercept = 1946, color = "red", size = 1, linetype = "dashed") +
  geom_vline(xintercept = 1996, color = "blue", size= 1, linetype = "dashed")
```

>Q3. Rates of pertussis increased after the aP vaccine. Possible reasons for this are: vaccine hesitancy, evolution of B. pertussis, increased testing, and a decreasing immunity among those vaccinated with the aP vaccine rather than the wP vaccine.

```{r}
library(jsonlite)
```

#Exploring CMI-PDB data
We'll use the **jsonlite** package to read from the CMI-PB database API directly.

```{r}
url <- "https://www.cmi-pb.org/api/subject"

subject <- read_json(url, simplifyVector =  TRUE)
head(subject, 3)
```

```{r}
table(subject$infancy_vac)
nrow(subject)
```

>Q4. ap: 47 wP: 49

```{r}
table(subject$biological_sex)
```

>Q5. Female: 66 Male: 30

```{r}
table(subject$biological_sex, subject$race)
```

>Q6. Female American Indian/Alaska Native: 0
      Female Asian: 18
      Female Black/African American: 2
      Female More Than One Race: 8
      Female Native Hawaiian/Other Pac. Islander: 1
      Female Unknown/Not Reported: 10
      Female White: 27
      Male American Indian/Alaska Native: 1
      Male Asian: 9
      Male Black/African American: 0
      Male More Than One Race: 2
      Male Native Hawaiian/Other Pac. Islander: 1
      Male Unknown/Not Reported: 4
      Male White: 13

```{r}
library(lubridate)
```

>Q7. $ Q8. optional

#Join datasets.

```{r}
specimen <- read_json("https://www.cmi-pb.org/api/specimen", simplifyVector = TRUE) 
titer <- read_json("https://www.cmi-pb.org/api/ab_titer", simplifyVector = TRUE)
```

Take a quick look. 

```{r}
head(specimen, 3)
```

I need to use inner_join() here.

>Q9. 

```{r}
library(dplyr)
library(tidyverse)
```

>Q9. 

```{r}
meta <- inner_join(specimen, subject)
dim(meta)
head(meta)
```

>Q10. 

```{r}
abdata <- inner_join(titer, meta)
dim(abdata)
```

```{r}
table(abdata$isotype)
```

>Q11. IgE: 6698  IgG:1413  IgG1: 6141 IgG2:6141 IgG3:6141 IgG4:  6141 
          

```{r}
table(abdata$visit)
```

>Q12. There are very vew visit 8 specimens compared to other visits. It's likely unfinished. 


Now we'll exclude visit 8 because it is unfinished.
```{r}
ig1 <- abdata %>% filter(isotype == "IgG1", visit!=8)
head(ig1)
```

>Q13. 

```{r}
ggplot(ig1) +
  aes(ab_titer, antigen) +
  geom_boxplot() + 
  facet_wrap(vars(visit), nrow=2)
```

>Q14. The FIM2/3 antigen has shifted. This is likely because antibodies have specifically recognized it. 

```{r}
ggplot(ig1) +
  aes(ab_titer, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(visit), nrow=2) +
  theme_bw()
```

```{r}
ggplot(ig1) +
  aes(ab_titer, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(infancy_vac, visit), nrow=2)
```

>Q15. 

```{r}
filter(ig1, antigen=="Measles") %>%
  ggplot() +
  aes(ab_titer, col=infancy_vac) +
  geom_boxplot(show.legend = TRUE) +
  facet_wrap(vars(visit)) +
  theme_bw()
```

```{r}
filter(ig1, antigen=="FIM2/3") %>%
  ggplot() +
  aes(ab_titer, col=infancy_vac) +
  geom_boxplot(show.legend = TRUE) +
  facet_wrap(vars(visit)) +
  theme_bw()
```
>Q16. The measles course is remarkable steady. It scarcely changes at all through the 8 visits. The FIM2/3 data, however, shows quite a lot of change. In both aP and wpP trials, it rises pretty consistently until visit 5, after which there is a slight decline.  
>Q17. No. 

# Pull RNA-Seq data from the CMI-PB database.
We can use the CMI-PB API to pull obtain time-course RNA-Seq results for wP and aP subjects (i.e. patients).


```{r}
url <- "https://www.cmi-pb.org/api/v2/rnaseq?versioned_ensembl_gene_id=eq.ENSG00000211896.7"
rna <- read_json(url, simplifyVector = TRUE) 
```

```{r}
ssrna <- inner_join(rna, meta)
dim(ssrna)
```

>Q18. 

```{r}
ggplot(ssrna) +
  aes(visit, tpm) +
  geom_point() 
```

```{r}
ggplot(ssrna) +
  aes(visit, tpm, group=subject_id) +
  geom_point() +
  geom_line(alpha=0.2)
```

>Q19. It's at its maximum at around visit 4. 
>Q20. It sort of matches. The AB Titer data suggested a peak at around visit 5 while the gene peaks at visit 4. The gene expression leads to the creation of antibodies; once a sufficient quantity of the antibody has been manufactured, the cell expression drops off. At this point, many antibodies are present (peaking at visit 5) and persist for some time. 

```{r}
ggplot(ssrna) +
  aes(tpm, col=infancy_vac) +
  geom_boxplot() +
  facet_wrap(vars(visit))
```





















